package idgs.tpc.pb;

message MigrationVerifyRequest {
  optional string schema_name = 2;
  optional string store_name  = 3;
}

message KeyPartition {
  required int32 key_partition_id = 1;
  required int32 key_member_id    = 2;
}

message PartitionMigrationData {
  required int32 partition_id         = 1;
  required int32 position             = 2;
  required int32 member_id            = 3;
  required uint32 size                = 4;
  repeated KeyPartition key_partition = 5;
}

message StoreMigrationData {
  required string schema_name                    = 1;
  required string store_name                     = 2;
  repeated PartitionMigrationData partition_data = 3;
}

message MemberMigrationData {
  required int32 member_id               = 1;
  repeated StoreMigrationData store_data = 2;
}

message MigrationVerifyResponse {
  required int32 result_code               = 1;
  repeated MemberMigrationData member_data = 2;
}

message SyncVerifyRequest {
  optional string schema_name = 2;
  optional string store_name  = 3;
}

message StoreSyncData {
  required string schema_name = 1;
  required string store_name  = 2;
  required uint64 data_size   = 3;
}

message MemberSyncData {
  required int32 member_id          = 1;
  repeated StoreSyncData store_data = 2;
}

message SyncVerifyResponse {
  required int32 result_code          = 1;
  repeated MemberSyncData member_data = 2;
}